oozie/src/main/java/org/apache/falcon/logging/JobLogMover.java

```diff
@@ -84,17 +85,23 @@ public void moveLog(WorkflowExecutionContext context){
 
     public int run(WorkflowExecutionContext context) {
         try {
-            OozieClient client = new OozieClient(context.getWorkflowEngineUrl());
+            String engineUrl = context.getWorkflowEngineUrl();
+            if (StringUtils.isBlank(engineUrl)) {
+                LOG.warn("Unable to retrieve workflow url for {} with status {} ",
+                        context.getWorkflowId(), context.getWorkflowStatus());
+                return 0;
```

Comment:
If the host has moved, the log movement anyway can't happen as the new host cannot return the old logs.